### DISCUSSION TITLE
Hono RPC isn't typesafe for queries

### DISCUSSION THREAD

**Discussion by adarshaacharya** [honojs/hono#3728] [Q&A]
I'm using hono backend with nextjs in FE

using rpc to communicate data between FE & BE

but found there are so many problems with query params.

- can't send number as query params
- can't send date as query params
- and most frustating can't send  object in query params

hono rpc expects everything to be send as string

Query params only accept object, but shouldn't this be handled by hono rpc itself. (type safety is really messed up)

can anyone suggest how to handle these cases

**Comment by lord007tn** [comment]
you should stringify them yourself 
by default queries are strings only same for params 
so thats an expected behaviour

**Comment by sudoskys** [comment]
You should not use Hono's RPC(Not out of any prejudice, I just provide a new idea), it has many hidden limitations. I recommend using monorepo to build common type definitions.

**Reply by chrisisler** [reply]
@sudoskys Could you list some of these hidden limitations? I'm implementing Hono on a project and am curious about any flaws that might have gone unnoticed in my evaluation of backend frameworks. Thanks

**Reply by sudoskys** [reply]
> @sudoskys Could you list some of these hidden limitations? I'm implementing Hono on a project and am curious about any flaws that might have gone unnoticed in my evaluation of backend frameworks. Thanks

Overall: reliability is definitely possible, but the development experience may not be very smooth (?)

The type system may not work well when you use multiple instances of Hono in combination. (You may need a magic crystal ball if this happens)

And if you use middleware, it's likely that you'll receive unintended unknown / any in some use cases.

(though as far as I know, Hono is working on improving this! https://github.com/honojs/hono/issues/2719#issuecomment-3248617207)
